execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1798,4 +1801,205 @@ where
             commit_result => Ok(commit_result),
         }
     }
+
+    pub fn commit_step(
+        &self,
+        correlation_id: CorrelationId,
+        step_request: StepRequest,
+    ) -> Result<StepResult, Error> {
+        let protocol_data = match self.state.get_protocol_data(step_request.protocol_version) {
+            Ok(Some(protocol_data)) => protocol_data,
+            Ok(None) => {
+                return Ok(StepResult::InvalidProtocolVersion);
+            }
+            Err(_) => {
+                return Ok(StepResult::PreconditionError);
+            }
+        };
+
+        let tracking_copy = match self.tracking_copy(step_request.parent_state_hash) {
+            Err(_) => return Ok(StepResult::PreconditionError),
+            Ok(None) => return Ok(StepResult::RootNotFound),
+            Ok(Some(tracking_copy)) => Rc::new(RefCell::new(tracking_copy)),
+        };
+
+        let executor = Executor::new(self.config);
+
+        let preprocessor = {
+            let wasm_costs = self
+                .wasm_costs(step_request.protocol_version)
+                .unwrap()
+                .unwrap();
+            Preprocessor::new(wasm_costs)
+        };
+
+        let auction_hash = protocol_data.auction();
+
+        let auction_contract = match tracking_copy
+            .borrow_mut()
+            .get_contract(correlation_id, auction_hash)
+        {
+            Ok(contract) => contract,
+            Err(_) => {
+                return Ok(StepResult::PreconditionError);
+            }
+        };
+
+        let auction_module = match tracking_copy.borrow_mut().get_system_module(
+            correlation_id,
+            auction_contract.contract_wasm_hash(),
+            self.config.use_system_contracts(),
+            &preprocessor,
+        ) {
+            Ok(module) => module,
+            Err(_) => {
+                return Ok(StepResult::PreconditionError);
+            }
+        };
+
+        if !self.system_contract_cache.has(auction_hash) {
+            self.system_contract_cache
+                .insert(auction_hash, auction_module.clone());
+        }
+
+        let system_account = Account::new(
```

Comment:
Small nit: I mildly prefer the the `Account::create` constructor that @mpapierski used in #265 :

```rust
        let virtual_system_account = {
            let named_keys = NamedKeys::new();
            let purse = URef::new(Default::default(), AccessRights::READ_ADD_WRITE);
            Account::create(SYSTEM_ACCOUNT_ADDR, named_keys, purse)
        };
```

Reply:
did this